frontend/src/App.test.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { shallow, configure } from "enzyme";
+import Adapter from 'enzyme-adapter-react-16';
+import { enableFetchMocks } from 'jest-fetch-mock';
+enableFetchMocks();
+import App from './App';
+
+configure({ adapter: new Adapter() });
+
+const LOGIN_URL = "/_ah/login?continue=%2F";
+const LOGOUT_URL = "/_ah/logout?continue=%2F";
+const LOGIN_MESSAGE = "logged in\n/_ah/logout?continue=%2F";
+const LOGOUT_MESSAGE = "stranger\n/_ah/login?continue=%2F";
```

Comment:
nit: prefer "/_ah/logout?continue=%2F" and "/_ah/login?continue=%2F" to be reused

ie in the const LOGIN_MESSAGE, you can have "logged in\n" + LOGIN_URL instead of adding that string constant again

(also what is the "_ah" refer to, just curious)